frontend/ui/widget/inputdialog.lua

```diff
@@ -456,6 +456,10 @@ function InputDialog:init()
     if self.fullscreen and not self.keyboard_visible then
         self:lockKeyboard(true)
     end
+
+    if Device:hasKeyboard() and G_reader_settings:isFalse("virtual_keyboard_enabled") then
```

Comment:
No need to check for `Device:hasKeyboard()`, you cannot toggle the setting without it.

Reply:
There's always the chance/case a user is migrating his settings between devices, and the setting would be there. Dunno how careful we are about that in other places.